include/base/default_coupling.h

```diff
@@ -50,10 +51,28 @@ class DefaultCoupling : public GhostingFunctor
 #ifdef LIBMESH_ENABLE_PERIODIC
     _periodic_bcs(nullptr),
 #endif
-    _mesh(nullptr),
     _n_levels(0)
   {}
 
+  /**
+   * Constructor.
+   */
+  DefaultCoupling(const DefaultCoupling & other) :
+    GhostingFunctor(other),
+    _dof_coupling(other._dof_coupling),
+#ifdef LIBMESH_ENABLE_PERIODIC
+    _periodic_bcs(other._periodic_bcs),
```

Comment:
On second glance, this shallow ```PeriodicBoundaries *``` copy (here and elsewhere) is an issue, isn't it?  other._periodic_bcs, if it exists, is pointing to the periodic boundary conditions associated with an old system, and if those get changed or if different PBCs get used with the new system, then we're out of sync with what we'd intended.

I don't think that should hold up this PR (which is at least a huge improvement, and should be all the improvement we need for the displaced mesh case in MOOSE) but it's at least a problem to puzzle over.

Reply:
I was thinking about this loudly easier. I should add some comments here. For the displaced system and the reference system, we are using the same PBC. I am not advocating it is a great idea. Here is some code from `AddPeriodicBCAction`

   ```
   if (component >= 0)
      {
        const std::pair<BoundaryID, BoundaryID> * boundary_ids =
            _mesh->getPairedBoundaryMapping(component);
        RealVectorValue v;
        v(component) = _mesh->dimensionWidth(component);
        PeriodicBoundary p(v);

        if (boundary_ids == NULL)
          mooseError(
              "Couldn't auto-detect a paired boundary for use with periodic boundary conditions");

        p.myboundary = boundary_ids->first;
        p.pairedboundary = boundary_ids->second;
        setPeriodicVars(p, getParam<std::vector<VariableName>>("variable"));
        nl.dofMap().add_periodic_boundary(p);
        if (displaced_problem)
          displaced_problem->nlSys().dofMap().add_periodic_boundary(p);
      }


```

I should put some comments here before we figure out an acceptable solution.

Reply:
Possibly need a clone for PBC?

Reply:
Looks like we already have a clone for PeriodicBoundary itself (IIRC the Akselos folks needed a subclass to do weirder transformations?), so it should be easy to add a whole PeriodicBoundaries::clone().

Trouble is, that might not be what we want.  It's at least not a complete solution.  Typically PeriodicBoundaries-aware functors are associated with the DofMap of the system with periodic variables, not with the underlying Mesh.  If we make a cloned copy of the PeriodicBoundaries associated with system1, then that'll solve the half of the problem where we don't want subsequent changes in system1's PBCs to affect ghosting functors on system2, but it won't solve the half of the problem where we do want subsequent changes in system2's PBCs to affect its ghosting functors, because the cloned PeriodicBoundaries won't be the same as the set that gets attached to system2 via it's DofMap::add_periodic_boundary().

Reply:
Yes, we have to leave it as-is for a moment. "as-is" (sharing PBC across systems) is what moose uses right now. 

But should we at least create an issue on this? We could revisit it when we think we are smart :-)

Reply:
[Issue created.](https://github.com/libMesh/libmesh/issues/2670)

> We could revisit it when we think we are smart :-)

Gonna need a time machine for that.

When I was 11 taking honors Algebra I, I thought I was smart.  Eventually that sort of thing got me to places where I wasn't  the smartest kid in the building, then to where I wasn't the smartest student in the class, then to where I wasn't the smartest researcher in the group ... at this point I find it safest to assume I'm not the smartest person in any given conversation.  It's most demoralizing that I haven't been able to make an exception for talking to my kids.

Reply:
LOL